### PR TITLE
DBSCAN kernel split: separate kernels for computation with and w/o we…

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

**Review by KulikovNikita — APPROVED** [review]

File: cpp/daal/src/algorithms/dbscan/oneapi/dbscan_kernel_ucapi.h [hunk 1]
```diff
@@ -46,8 +46,10 @@ class DBSCANBatchKernelUCAPI : public Kernel
                              const Parameter * par);
 
 private:
-    services::Status getCores(const services::internal::sycl::UniversalBuffer & data, uint32_t nRows, uint32_t nFeatures, algorithmFPType nNbrs,
+    services::Status getCores(const services::internal::sycl::UniversalBuffer & data, uint32_t nRows, uint32_t nFeatures, int nNbrs,
                               algorithmFPType eps);
+    services::Status getCoresWithWeights(const services::internal::sycl::UniversalBuffer & data, uint32_t nRows, uint32_t nFeatures,
+                                         algorithmFPType nNbrs, algorithmFPType eps);
```
[the remark this answers is not in the document]
Comment (amgrigoriev) [reply cpp/daal/src/algorithms/dbscan/oneapi/dbscan_kernel_ucapi.h:52]: It's a key point for optimization

File: cpp/daal/src/algorithms/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i [hunk 2]
```diff
@@ -311,6 +311,45 @@ services::Status DBSCANBatchKernelUCAPI<algorithmFPType>::getCores(const Univers
...
+template <typename algorithmFPType>
+services::Status DBSCANBatchKernelUCAPI<algorithmFPType>::getCoresWithWeights(const UniversalBuffer & data, uint32_t nRows, uint32_t nFeatures,
+                                                                              algorithmFPType nNbrs, algorithmFPType eps)
+{
+    services::Status st;
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.getCores);
+    auto & context        = Environment::getInstance()->getDefaultExecutionContext();
+    auto & kernel_factory = context.getClKernelFactory();
+    DAAL_CHECK_STATUS_VAR(buildProgram(kernel_factory));
+    auto kernel = kernel_factory.getKernel("computeCoresWithWeights", &st);
```
[the remark this answers is not in the document]
Comment (amgrigoriev) [reply cpp/daal/src/algorithms/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i:353]: getKernel change hasn't been committed yet

**Pull request merged**